Spring-basics-tutorial/src/main/java/com/vinnik/kate/topic/TopicRepository.java

```diff
@@ -0,0 +1,6 @@
+package com.vinnik.kate.topic;
+
+import org.springframework.data.repository.CrudRepository;
+
+public interface TopicRepository extends CrudRepository<Topic, String> {
```

Comment:
Please, explain a purpose of defining an empty interface in this case.

Reply:
I read that empty interfaces also called marker interfaces and might be useful in developing API, like read in this answer: https://javarevisited.blogspot.com/2012/01/what-is-marker-interfaces-in-java-and.html . It is alternative to annotations in spring, as I understood.
Creating marker interfaces is a good way to classify code: it describes specific behavior on the part of the implementing class (https://en.wikipedia.org/wiki/Marker_interface_pattern)

It seems to me, that in my case it was implemented to underline creating my own TopicRepository. Creating CrudRepository in this case would be a mistake, since CrudRepository is another level of abstraction and doesn't says nothing about its role in program, but TopicRepository says and might be supplemented with new methods in future.